### ISSUE TITLE
Replication Parameters and How to use MixStyle and EFDM

### ISSUE THREAD

**Issue by VirtueZhao** [KaiyangZhou/Dassl.pytorch#50]
Dear Kaiyang

Really appreciate the open-source domain generalization framework. It is really amazing.
I'm currently working on replicating the results and extending my work on the current framework. Do you mind sharing the parameters for each baseline to replicate the results mentioned in your AAAI-2020 (DDAIG)? When I tried to replicate DDAIG, I found that the accuracy was good but could not achieve the results mentioned in the paper. On the other hand, the baseline results of CrossGrad, DAELDG, and DomainMix are much worse than the paper. I guess it may be caused by the parameter tunning because I'm currently using the default settings of the framework (I didn't change any single line of the framework).

On the other hand, could you please post an instruction about how to run MixStyle and EFDM by the framework?

Thank you very much!

**Comment by KaiyangZhou** [comment]
To reproduce DDAIG, please refer to the log files in https://github.com/KaiyangZhou/Dassl.pytorch/issues/6#issue-653895977.

CrossGrad's hyper-parameters should be these https://github.com/KaiyangZhou/Dassl.pytorch/blob/master/dassl/config/defaults.py#L260 (but forgot if I had tuned them).

I wasn't the one submitting the DomainMix code. Perhaps you can check with the author who submitted the PR.

The example of running MixStyle or EFDMix on, e.g., pacs, is
```python
python tools/train.py \
--root ${DATA_ROOT} \
--seed 1 \
--trainer Vanilla \  # aka ERM
--dataset-config-file configs/datasets/dg/pacs.yaml \  # the standard params for the pacs dataset
--config-file configs/trainers/dg/vanilla/pacs.yaml \  # the custom params for vanilla
--source-domains art_painting cartoon photo \
--target-domains sketch \
--output-dir ${OUTPUT_DIR} \
MODEL.BACKBONE.NAME resnet18_ms_l12  # this will override the model name in configs/datasets/dg/pacs.yaml
```

To use Dassl as a library, I'd suggest you take a look at the structure of https://github.com/KaiyangZhou/mixstyle-release/tree/master/imcls and https://github.com/KaiyangZhou/CoOp. Don't need to dig deep into the code. Just follow the structure. So basically you need to 1) have `train.py` and add your trainer's config variables, 2) create folders for `trainers`, `datasets` (if you have new datasets), or `models` (if you use custom models not provided in this repo).

Hope this helps.
